### ISSUE TITLE
OCP 4.12: some tests failed with PodSecurity restriction errors: "restricted:v1.24": host namespaces (hostNetwork=true)

### ISSUE THREAD

**Issue by maayanf24** [submariner-io/submariner-operator#2330]
**What happened**:
deploy with `subctl` on top of OCP 4.12 clusters using `0.14.0-rc2`
Some E2E tests failed, full log attached.

```
[dataplane] Basic TCP connectivity tests across clusters without discovery
github.com/submariner-io/submariner@v0.14.0-rc2/test/e2e/dataplane/tcp_pod_connectivity.go:28
  when a pod with HostNetworking connects via TCP to a remote pod
  github.com/submariner-io/submariner@v0.14.0-rc2/test/e2e/dataplane/tcp_pod_connectivity.go:109
    when the pod is on a gateway and the remote pod is not on a gateway
    github.com/submariner-io/submariner@v0.14.0-rc2/test/e2e/dataplane/tcp_pod_connectivity.go:121
      should have sent the expected data from the pod to the other pod [It]
      github.com/submariner-io/submariner@v0.14.0-rc2/test/e2e/dataplane/tcp_pod_connectivity.go:36

      Unexpected error:
          <*errors.StatusError | 0xc00047b680>: {
              ErrStatus: {
                  TypeMeta: {Kind: "", APIVersion: ""},
                  ListMeta: {
                      SelfLink: "",
                      ResourceVersion: "",
                      Continue: "",
                      RemainingItemCount: nil,
                  },
                  Status: "Failure",
                  Message: "pods \"tcp-check-podn8t9f\" is forbidden: violates PodSecurity \"restricted:v1.24\": host namespaces (hostNetwork=true)",
                  Reason: "Forbidden",
                  Details: {
                      Name: "tcp-check-podn8t9f",
                      Group: "",
                      Kind: "pods",
                      UID: "",
                      Causes: nil,
                      RetryAfterSeconds: 0,
                  },
                  Code: 403,
              },
          }
          pods "tcp-check-podn8t9f" is forbidden: violates PodSecurity "restricted:v1.24": host namespaces (hostNetwork=true)
      occurred

      github.com/submariner-io/shipyard@v0.14.0-rc2/test/e2e/framework/network_pods.go:343
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Diagnose information (use `subctl diagnose all`):
- subctl diagnose firewall inter-cluster --context cluster-a1 --remotecontext cluster-b1
 ✗ Checking if tunnels can be setup on the gateway node of cluster "maafried-cluster-a1" 
 ✗ Error: the gateway on cluster "maafried-cluster-b1" does not have an active connection to cluster "maafried-cluster-a1"
 ✗ Error retrieving the gateway IP of cluster "maafried-cluster-a1": Error: the gateway on cluster "maafried-cluster-b1" does not have an active connection to cluster "maafried-cluster-a1"
 ✗ Could not determine if Tunnels can be established on the gateway node of cluster "maafried-cluster-a1"

subctl version: v0.14.0-rc2

- Gather information (use `subctl gather`)
- Cloud provider or hardware configuration:
- Install tools:
- Others:

**Comment by maayanf24** [comment]
[maafried-cluster-a1.tar.gz](https://github.com/submariner-io/submariner-operator/files/9950044/maafried-cluster-a1.tar.gz)
[maafried-cluster-b1.tar.gz](https://github.com/submariner-io/submariner-operator/files/9950054/maafried-cluster-b1.tar.gz)
[testday_0.14.0-rc2-full-e2e.txt](https://github.com/submariner-io/submariner-operator/files/9949846/testday_0.14.0-rc2-full-e2e.txt)

**Comment by skitt** [comment]
The `diagnose` error is caused by the mismatch between context names and cluster names. This is tracked with https://github.com/submariner-io/submariner-operator/issues/2333. This issue will address the `PodSecurity` errors.

**Comment by tpantelis** [comment]
From [this blog](https://cloud.redhat.com/blog/managing-sccs-in-openshift), I think we're going to have to use the `hostnetwork` SCC (at least). It seems we'll have to create a `ServiceAccount` in the e2e namespace and associate to the desired SCC via a cluster role binding as described in the blog and as we do in [subctl](https://github.com/submariner-io/subctl/blob/abd0710c96fa1d2a637d00782b3744dbd7336348/pkg/operator/ensure.go#L68).

Another option is to use the `submariner-operator` namespace which is already set up with a privileged-level default SA (it seems) but we'd have to ensure all the tests clean up the resources they create.

**Comment by tpantelis** [comment]
In shipyard, we create the test namespace with `enforce` pod security label set to `privileged`, eg:

```
metadata:
  generateName: e2e-tests-dataplane-conn-nd-
  labels:
    e2e-framework: dataplane-conn-nd
    pod-security.kubernetes.io/audit: privileged
    pod-security.kubernetes.io/enforce: privileged
    pod-security.kubernetes.io/warn: privileged
```

However, the resulting namespace has the label set to `restricted`:

```
metadata:
  annotations:
    openshift.io/sa.scc.mcs: s0:c40,c10
    openshift.io/sa.scc.supplemental-groups: 1001580000/10000
    openshift.io/sa.scc.uid-range: 1001580000/10000
  creationTimestamp: "2022-11-10T17:51:53Z"
  generateName: e2e-tests-dataplane-conn-nd-
  labels:
    e2e-framework: dataplane-conn-nd
    kubernetes.io/metadata.name: e2e-tests-dataplane-conn-nd-s8mvp
    pod-security.kubernetes.io/enforce: restricted
    pod-security.kubernetes.io/enforce-version: v1.24
```

So it seems OCP silently disallows `privileged` access although it's somehow allowed for the `submariner-operator` namespace.  I even tried changing the label manually via CLI and it succeeds but then is changed back to `restricted`.

**Comment by tpantelis** [comment]
> So it seems OCP silently disallows privileged access although it's somehow allowed for the submariner-operator > namespace. I even tried changing the label manually via CLI and it succeeds but then is changed back to restricted.

So this behavior is controlled by the OCP [label synchronization controller](https://github.com/openshift/enhancements/blob/master/enhancements/authentication/pod-security-admission-autolabeling.md). This behavior can be disabled by setting the `security.openshift.io/scc.podSecurityLabelSync` label to false. I added this label to a namespace and then changed the `enforce` label to `privileged` and it took. After that I was able to successfully create a network pod with host networking. 

I made the change in shipyard and the E2E tests succeeded!
